packages/dmn-editor/tests-e2e/zoomDiagram.spec.ts

```diff
@@ -0,0 +1,77 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { expect, test } from "./__fixtures__/base";
+import { NodeType } from "./__fixtures__/nodes";
+import { Tab } from "./__fixtures__/tabs";
+
+test.beforeEach(async ({ editor }) => {
+  await editor.open();
+});
+
+test.describe("Zoom and Panning Diagram", () => {
+  test("should zoom in diagram", async ({ nodes, palette, diagram, multipleNodesPropertiesPanel }) => {
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 100, y: 10 } });
+    await diagram.resetFocus();
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 100, y: 110 } });
+    await diagram.resetFocus();
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 100, y: 210 } });
+    await diagram.zoomIn({ clicks: 1 });
+    await expect(diagram.get()).toHaveScreenshot("zoom-in-diagram.png");
+  });
+
+  test("should zoom out diagram", async ({ nodes, palette, diagram, multipleNodesPropertiesPanel }) => {
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 100, y: 10 } });
+    await diagram.resetFocus();
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 100, y: 110 } });
+    await diagram.resetFocus();
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 100, y: 210 } });
+    await diagram.zoomOut({ clicks: 2 });
+    await expect(diagram.get()).toHaveScreenshot("zoom-out-diagram.png");
+  });
+
+  test("should fit to view diagram", async ({ nodes, palette, diagram, multipleNodesPropertiesPanel }) => {
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 10, y: 10 } });
+    await diagram.resetFocus();
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 400, y: 400 } });
+    await diagram.resetFocus();
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 200, y: 200 } });
+    await diagram.zoomIn({ clicks: 2 });
+    await diagram.fitView();
+    await expect(diagram.get()).toHaveScreenshot("fit-to-view-diagram.png");
+  });
+
+  test("should keep view settings after swap tabs", async ({ palette, diagram, tabs, page }) => {
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 100, y: 100 } });
+    await diagram.resetFocus();
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 200, y: 200 } });
+    await diagram.resetFocus();
+    await palette.dragNewNode({ type: NodeType.DECISION, targetPosition: { x: 200, y: 300 } });
+    await diagram.zoomOut({ clicks: 3 });
+    await page.mouse.move(500, 500);
+    await page.mouse.down();
+    await page.mouse.move(300, 300);
+    await page.mouse.up();
```

Comment:
If you're trying to pan the viewport, the middle mouse button should be used, right?

```suggestion
    await page.mouse.move(500, 500);
    await page.mouse.down({ button: "middle" });
    await page.mouse.move(300, 300);
    await page.mouse.up({ button: "middle" });
```

Reply:
In fact, it works even without the middle button, if you click in an area where there's nothing there.
But I accepted your suggestion because it made it more clear what is happening and also added a comment above. Thank you!